src/java.base/share/classes/java/lang/reflect/ProxyGenerator.java

```diff
@@ -952,65 +924,66 @@ public String toString() {
     }
 
     /**
-     * A PrimitiveTypeInfo object contains assorted information about
-     * a primitive type in its public fields.  The struct for a particular
+     * A PrimitiveTypeInfo object contains bytecode-related information about
+     * a primitive type in its instance fields. The struct for a particular
      * primitive type can be obtained using the static "get" method.
      */
-    private static class PrimitiveTypeInfo {
-
-        private static Map<Class<?>, PrimitiveTypeInfo> table = new HashMap<>();
-
-        static {
-            add(byte.class, Byte.class);
-            add(char.class, Character.class);
-            add(double.class, Double.class);
-            add(float.class, Float.class);
-            add(int.class, Integer.class);
-            add(long.class, Long.class);
-            add(short.class, Short.class);
-            add(boolean.class, Boolean.class);
-        }
+    private static final class PrimitiveTypeInfo {
+        private static final PrimitiveTypeInfo BYTE = new PrimitiveTypeInfo(byte.class, 0);
```

Comment:
Can this be `private enum PrimitiveTypeInfo...` or perhaps a record class?

Reply:
Thanks. I've converted it into an enum following the suit of `sun.invoke.util.Wrapper`. I think an enum fits better here as we don't need the canonical constructor and object methods of a record.